src/app/server/AppDelegate.h

```diff
@@ -29,11 +29,31 @@ class AppDelegate
 public:
     virtual ~AppDelegate() {}
     /**
-     * This is called on start of session establishment process
+     * This is called when the PBKDFParamRequest is received and indicates the start of the session establishment process
      */
     virtual void OnCommissioningSessionEstablishmentStarted() {}
+
+    /**
+     * This is called when the commissioning session has been established
+     */
     virtual void OnCommissioningSessionStarted() {}
-    virtual void OnCommissioningSessionStopped(CHIP_ERROR err) {}
+
+    /**
+     * This is called when there is an error in establishing a commissioning session (such as, when an invalid passcode is provided)
+     *
+     * @param err CHIP_ERROR indicating the error that occurred during session establishment
+     *
+     * @return true if the commissioning window should be closed, false otherwise
```

Comment:
I don't believe it's spec-compliant for a commissionee to randomly close a commissioning window (esp. one opened with OpenCommissioningWindow) until either the max failed attempts is hit or the timeout has been reached...  Why are we adding ways to do non-spec-compliant things?

Reply:
@tcarmelveilleux can you help answer this question?

Reply:
@bzbarsky-apple If you can show me in the spec where it's *forbidden* for a device to close a commissioning window (e.g. via UI or via less errors than the max 20), I would love to see it.

Reply:
What the spec has is:

> This field SHALL specify the time in seconds during which commissioning session establishment is allowed by the Node.

and

> When a Node receives the Open Commissioning Window command, it SHALL begin advertising on DNS-SD as described in Section 4.3.1, “Commissionable Node Discovery” and for a time period as described in Section 11.18.8.1.1, “CommissioningTimeout Field”. 

There are then explicit provisions for RevokeCommissioning to close the commissioning window and for CommissioningComplete to close the commissioning window and for the 20-error situation.  Ideally the statements above would have made it clear that it's "shall for this time except if something says otherwise", but as things stand we have normative statements that commissioning session establishment SHALL be allowed and advertising SHALL continue for the provided period of time, with some other statements that contradict that in specific cases.

Reply:
Let's remove the bool: it's not worth arguing over, since it's currently ambigous. Let's not forcibly close on true.

Reply:
Reverted the introduction of the bool.